### ISSUE TITLE
Backward Compatibility (1.3.3)

### ISSUE THREAD

**Issue by andrewio** [yasirkula/UnityNativeCamera#99]
Hi, I just updated from 1.3.3 to 1.3.7 and got next:

- error CS7036: There is no argument given that corresponds to the required formal parameter 'isPicturePermission' of 'NativeCamera.RequestPermission(bool)'
- error CS7036: There is no argument given that corresponds to the required formal parameter 'isPicturePermission' of 'NativeCamera.CheckPermission(bool)'

It seems that you added a parameter, without the default value. 

I use your Asset in many of my assets on Asset Store, so please add the default value to this parameter (true or false) to have a backward compatibility and reduce negative emails to my support =)

**Comment by yasirkula** [comment]
I understand the problem. But the default parameter for CheckPermission wouldn't make sense here because we have to know how the user is planning to use the camera so that we can ask the right permission (I'm not a fan of the changes to permission system either but Android & iOS make big changes to the permission system from time to time and all I can do is adapting to those changes). Also, if we assume that the default value checks for picture permission, then it wouldn't work for you anyways because you're attempting to get the video permission.

Since your use-case is only asking for permissions, you can use [AndroidRuntimePermissions](https://github.com/yasirkula/UnityAndroidRuntimePermissions) or simply Unity's own Permission API to request camera permission on Android. On iOS, feel free to copy the permission related functions from my native code (a link to this repo in the native code would be appreciated but not needed). This way, you can eliminate the NativeCamera dependency. I can't recommend telling your users to use a previous release of NativeCamera because Android made a change in the permission system and thus, the previous releases won't work on Android 13 and newer (this issue was apparent in numerous plugins of mine but I've updated them all).

**Comment by andrewio** [comment]
>  I checked the both values: true or false with AR Face Tracking. And the video stream works in the both cases.
Android 13, Galaxy A71, Unity 2021.3.16.

**Comment by yasirkula** [comment]
Yes because you're checking the camera permission only :D That permission is common in both cases. What changes is whether READ_MEDIA_IMAGES or READ_MEDIA_VIDEO is asked. It used to be READ_EXTERNAL_STORAGE only so a bool parameter wasn't necessary but now Android decided to split up that permission.
